### ISSUE TITLE
How to use streaming response ?

### ISSUE THREAD

**Issue by sydneyfunnelAIO** [lostisland/faraday#1071]
## Basic Info

* Faraday Version:1.0 --pre
* Ruby Version:2.5.6

## Issue description
hello i am codding a chatbot for facebook live. and I need response live comments with a link.
yeah i know u sends me link: https://lostisland.github.io/faraday/usage/streaming
but i didnt understand how must ı use it. i tried thinks. can u send me any example for that.
i am newbie on faraday[trying just for this project].
thx again... 

## Steps to reproduce
![image](https://user-images.githubusercontent.com/51324523/68312013-7a60ac00-00c3-11ea-8a95-3e46181c1768.png)

**Comment by iMacTia** [comment]
Hi @sydneyfunnelAIO, not sure how used you are with Ruby, but basically `req.options.on_data` is a callback function that will be executed every time you receive a chunk of data.
A good starting point, would be to just print what you receive:

```ruby
fb = 'https://streaming-graph.facebook.com'
conn = Faraday.new(fb)
conn.get('/9822...') do |req|
  req.options.on_data = Proc.new { |chunk, size| puts chunk }
end
```

If this works, you can then go on and change `req.options.on_data` to do whatever you want with `chunk` and `size`.

I hope this helps, please let me know how that goes

**Comment by sydneyfunnelAIO** [comment]
i tried but its didnt response anything. 
![image](https://user-images.githubusercontent.com/51324523/68399975-5024f200-0188-11ea-8939-908da81eb9ac.png)
my command line:
![image](https://user-images.githubusercontent.com/51324523/68400046-6f238400-0188-11ea-8003-8987ba1967d3.png)

**Comment by iMacTia** [comment]
It would be easier to help if I could try this out, however I'm not a FB developer.
Is it possible to get a temporary `access_token` that I could use to try this locally?
You can send it over to me privately on Gitter

**Comment by snltd** [comment]
@sydneyfunnelAIO how long are you leaving the connection open before killing it? 

The problem I'm having -- possibly the same as you -- is that I get nothing for a long time, then suddenly a lot of data all at once. Hitting the same endpoint with curl produces a steady stream of data.

I haven't yet worked out why I see this pattern.

**Comment by iMacTia** [comment]
For everyone's info: I've spent some time investigating this with @sydneyfunnelAIO and we realised that even taking Faraday out of the equation and using plain `Net::HTTP`, the issue persists.
We then tried calling another streaming response with the same code snippet:

```ruby
conn = Faraday.new('https://postman-echo.com')
conn.get('/stream/500') do |req|
  req.options.on_data = Proc.new do |chunk, size|
    puts chunk
    sleep 1
  end
end
```

and it worked exactly as expected. So we agreed there's something off with the Facebook API and this is not an issue in Faraday.
